pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -177,10 +214,154 @@ func buildRestConfig(clusterURL string, bearerToken string) *rest.Config {
 
 // buildClient builds a controller-runtime client for interacting with
 // a K8s cluster.
-func buildClient(config *rest.Config) (k8sclient.Client, error) {
+func (m *MigCluster) buildClient(config *rest.Config) (k8sclient.Client, error) {
 	c, err := k8sclient.New(config, k8sclient.Options{Scheme: scheme.Scheme})
 	if err != nil {
 		return nil, err
 	}
 	return c, nil
 }
+
+// Delete resources on the cluster by label.
+func (m *MigCluster) DeleteResources(client k8sclient.Client, labels map[string]string) error {
+	client, err := m.GetClient(client)
+	if err != nil {
+		return err
+	}
+	if labels == nil {
+		labels = map[string]string{PartOfLabel: Application}
+	}
+
+	options := k8sclient.MatchingLabels(labels)
+
+	// Secret
+	sList := kapi.SecretList{}
+	err = client.List(context.TODO(), options, &sList)
+	if err != nil {
+		return err
+	}
+	for _, r := range sList.Items {
+		err = client.Delete(context.TODO(), &r)
+		if err != nil && !k8serror.IsNotFound(err) {
+			return err
+		}
+	}
+
+	// Pod
+	pList := kapi.PodList{}
```

Comment:
I wonder whether we should delete pods later. Deleting them first might result in deployments and dcs starting to respawn them while we're trying to delete the dc.

Reply:
This is intended to deal with _stage_ Pods, not Pods related to the _registry_.  The migration does good job keeping them cleaned up but thinking there could be cases where they're _leaked_.

Reply:
I changed the order so that Deployment and DeploymentConfig get deleted before Pods to minimize thrashing.